scripts/ci_checks/macos/macos.sh

```diff
@@ -3,10 +3,14 @@
 set -euxo pipefail
 
 brew install \
-     automake scons ragel gengetopt\
-     libuv speexdsp sox \
+     automake scons ragel gengetopt \
+     libuv speexdsp sox openssl@3 \
      cpputest google-benchmark
 
+# > openssl@3 is keg-only, which means it was not symlinked into /usr/local,
+# > because macOS provides LibreSSL.
+export PKG_CONFIG_PATH="/usr/local/opt/openssl@3/lib/pkgconfig:${PKG_CONFIG_PATH:-}"
```

Comment:
As discussed in char, we should remove this and adjust build system to work with openssl from brew out of the box.

In `3rdparty/SConscript`, we should check that:

* we're running on macos and are not cross-compiling
* `brew` is in PATH (see env.Which)
* `brew --prefix` returns existing directory
* openssl is installed at `<PREFIX>/Cellar/openssl@<SLOT>/<VERSION>`

Where:

* PREFIX is output of `brew --prefix`
* SLOT is short version, e.g. `1.1` or `3`
* VERSION is full version, e.g. `1.1.1s`

I think we should not hard-code SLOT and VERSION, but instead see what is present and use the latest one.

If all these check succeeded, we can append `<PREFIX>/Cellar/openssl@<SLOT>/<VERSION>/lib/pkgconfig` to `PKG_CONFIG_PATH` and try to find openssl.

What we'll have in result:

* if user did not specify any option, we'll automatically use openssl 1.1 or 3 from brew
* user still can install openssl manually and set PKG_CONFIG_PATH to use it

Here is some output on one of my machines.

```
$ brew --prefix
/usr/local
```

```
$ tree -L 2 /usr/local/Cellar/openssl@1.1/1.1.1s
/usr/local/Cellar/openssl@1.1/1.1.1s
├── AUTHORS
├── CHANGES
├── INSTALL_RECEIPT.json
├── LICENSE
├── NEWS
├── README
├── bin
│   ├── c_rehash
│   └── openssl
├── include
│   └── openssl
├── lib
│   ├── engines-1.1
│   ├── libcrypto.1.1.dylib
│   ├── libcrypto.a
│   ├── libcrypto.dylib -> libcrypto.1.1.dylib
│   ├── libssl.1.1.dylib
│   ├── libssl.a
│   ├── libssl.dylib -> libssl.1.1.dylib
│   └── pkgconfig
└── share
    ├── doc
    └── man
```

```
$ ls -1 /usr/local/Cellar/openssl@1.1/1.1.1s/lib/pkgconfig
libcrypto.pc
libssl.pc
openssl.pc
```

Reply:
agreed to ignore SLOT and VERSION and instead use `$(bre --prefix)/opt/openssl` which is a link to the particular version, e.g. `$(brew --prefix)/opt/openssl@1.1/1.1.1s/`

Reply:
done